src/pages/workspace/distanceRates/CategorySelector/CategorySelectorModal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {useMemo} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Modal from '@components/Modal';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import useThemeStyles from '@hooks/useThemeStyles';
+import CONST from '@src/CONST';
+import type * as OnyxTypes from '@src/types/onyx';
+import type CategoryItemType from './types';
+
+type CategorySelectorModalProps = {
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<OnyxTypes.PolicyCategories>;
+
+    /** Whether the modal is visible */
+    isVisible: boolean;
+
+    /** Selected category  */
+    currentCategory: string;
+
+    /** Function to call when the user selects a category */
+    onCategorySelected: (value: CategoryItemType) => void;
+
+    /** Function to call when the user closes the category selector modal */
+    onClose: () => void;
+
+    /** Label to display on field */
+    label: string;
+};
+
+function CategorySelectorModal({policyCategories, isVisible, currentCategory, onCategorySelected, onClose, label}: CategorySelectorModalProps) {
```

Comment:
NAB we'll have to start loading categories on OpenPolicyDistanceRatesPage. This is backend work - [draft PR](https://github.com/Expensify/Auth/pull/10244).